blocks/form/form.js

```
@@ -3,10 +3,12 @@ import { sampleRUM } from '../../scripts/lib-franklin.js';
 function constructPayload(form) {
   const payload = {};
   [...form.elements].forEach((fe) => {
-    if (fe.type === 'checkbox') {
-      if (fe.checked) payload[fe.id] = fe.value;
-    } else if (fe.id) {
-      payload[fe.id] = fe.value;
+    if ( fe.type === "radio") {
+        if (fe.checked) payload[fe.name] = fe.value;
+    } else if (fe.type === "checkbox") {
+      if (fe.checked) payload[fe.name] = payload[fe.name] ? `${payload[fe.name]},${fe.value}`: fe.value
+    } else if (fe.name) {
+      payload[fe.name] = fe.value;
```

the check for fe.name should be the first check

```
if (fe.name) {
   if (fe.type === 'radio' && fe.checked) {
         payload[fe.name] = fe.value
   } else if (fe.type === 'checkox' && fe.checked) { .... }
   else {
       ...
   }
}
```

---

@vdua  i think name field is required for every input field, so ideally it should fail if name is undefined ? please correct if my understanding is correct

---

lint issues are fixed with form.js file. varun , i am seeing multiple lint issues as well with other blocks. i am using  airbnb-base-eslint config.

---

> i think name field is required for every input field
No that is not correct. Consider buttons, output, etc. elements. 

I was just saying that check for `fe.name` that is there in [Line 10](https://github.com/adobe/afb/pull/8/files#diff-14d0faad899dcfcb4e5cd318658f4d58b9e2aa66ebc5912bbe09302339f6a564R10) should be moved up.